boring/src/ssl/callbacks.rs

```diff
@@ -48,6 +48,30 @@ where
     verify(preverify_ok != 0, ctx) as c_int
 }
 
+pub(super) unsafe extern "C" fn raw_custom_verify<F>(
+    ssl: *mut ffi::SSL,
+    al: *mut u8,
+) -> ffi::ssl_verify_result_t
+where
+    F: Fn(&mut SslRef, &mut u8) -> bool + 'static + Sync + Send,
```

Comment:
Worth making API safer here by making the closure accept a mut ref to https://docs.rs/boring/latest/boring/ssl/struct.SslAlert.html instead of raw u8